File: ruby/rock_paper_scissors/rock_paper_scissors_8.rb
# This program is a simple implementation of the classic game "Rock, Paper, Scissors".
# The purpose of this program is to provide a basic example of how to use conditionals,
# loops, and user input in Ruby. The game allows a user to play against the computer,
# which randomly selects one of the three options. The program then determines the winner
# based on the standard rules of the game.

# Define the options available in the game
OPTIONS = ["rock", "paper", "scissors"]

# Method to get the computer's choice
def computer_choice
    OPTIONS.sample
end

# Method to get the user's choice
def user_choice
    puts "Please enter your choice (rock, paper, or scissors):"
    gets.chomp.downcase
end

# Method to determine the winner
def determine_winner(user, computer)
    if user == computer
        "It's a tie!"
    elsif (user == "rock" && computer == "scissors") ||
                (user == "paper" && computer == "rock") ||
                (user == "scissors" && computer == "paper")
        "You win!"
    else
        "Computer wins!"
    end
end

# Main game loop
loop do
    user = user_choice
    computer = computer_choice

    puts "You chose: #{user}"
    puts "Computer chose: #{computer}"

    puts determine_winner(user, computer)

    puts "Do you want to play again? (yes or no)"
    break if gets.chomp.downcase != "yes"
end

File: vbnet/rock_paper_scissors/rock_paper_scissors_8.vb
' This program is a simple implementation of the Rock-Paper-Scissors game in Visual Basic .NET.
' The purpose of this program is to demonstrate basic programming concepts such as:
' - Taking user input
' - Generating random choices
' - Using conditional statements to determine the winner
' - Displaying the result to the user

Module RockPaperScissors
    Sub Main()
        ' Declare variables to store user choice, computer choice, and the result
        Dim userChoice As String
        Dim computerChoice As String
        Dim result As String

        ' Display instructions to the user
        Console.WriteLine("Welcome to Rock-Paper-Scissors!")
        Console.WriteLine("Please enter your choice (rock, paper, or scissors):")

        ' Read user input
        userChoice = Console.ReadLine().ToLower()

        ' Generate a random choice for the computer
        Dim rand As New Random()
        Dim computerChoiceNumber As Integer = rand.Next(1, 4)

        ' Convert the random number to a choice (1 = rock, 2 = paper, 3 = scissors)
        Select Case computerChoiceNumber
            Case 1
                computerChoice = "rock"
            Case 2
                computerChoice = "paper"
            Case 3
                computerChoice = "scissors"
        End Select

        ' Determine the result of the game
        If userChoice = computerChoice Then
            result = "It's a tie!"
        ElseIf (userChoice = "rock" And computerChoice = "scissors") Or
               (userChoice = "paper" And computerChoice = "rock") Or
               (userChoice = "scissors" And computerChoice = "paper") Then
            result = "You win!"
        Else
            result = "You lose!"
        End If

        ' Display the computer's choice and the result
        Console.WriteLine("Computer chose: " & computerChoice)
        Console.WriteLine(result)

        ' Wait for user input before closing the program
        Console.WriteLine("Press any key to exit.")
        Console.ReadKey()
    End Sub
End Module

